### DISCUSSION TITLE
Suspect for a bug: Unintended Feature

### DISCUSSION THREAD

**Discussion by SPyofgame** [squidfunk/mkdocs-material#5210] [Q&A] [answered]
Where: mkdocs material >> setup >> changing the colors
Link: https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/#custom-colors
Issue: When I click the tile, instead of trying to change the color of the content, it changes for its next content

Action: I click on the "slate" tile
![image](https://user-images.githubusercontent.com/62207095/225986545-c926cb34-ab5c-42ed-a097-f3ffa2f86061.png)

Expected Result: I think it is intended to do something like this
![image](https://user-images.githubusercontent.com/62207095/225986634-e5fa20cd-0522-44c0-b459-6e60d1c08f76.png)

Result: But instead, I got this (the below content is changed)
![image](https://user-images.githubusercontent.com/62207095/225987263-e238d309-555e-4426-8e62-24aabc32cbfc.png)

It also affects the rest of the tiles in the following way:
- Click the tile from "color scheme", but "primary color" changed
- Click the tile from "primary color", but "accent color" changed
- Click the tile from "accent color", but "color palette toggle" changed

**Comment by ile-24556** [comment]
The value-changing scripts specify their targets by the IDs of the `pre`s. Normally, the IDs start with `__code_1` in a page, even in the cloned official site, and [the scripts expect that]. However, the IDs in the real official site start with `__code_0` and scripts' targets are shifted. 

| The official site | Cloned site | Other site |
| --- | --- | --- |
| ![image](https://user-images.githubusercontent.com/110268848/226085385-5f021006-7e5e-477e-9d06-358cc88c64ce.png) | ![image](https://user-images.githubusercontent.com/110268848/226085364-d4eeabbe-e841-41e0-a1bf-fd55c09af714.png) | ![image](https://user-images.githubusercontent.com/110268848/226085426-cde2b31b-9dbf-4f86-b542-36ebb41dbe1b.png) |

[the scripts expect that]: https://github.com/squidfunk/mkdocs-material/blob/4ad2dc1e288a0ffe78a8db3a04978a6e089ce35e/docs/setup/changing-the-colors.md?plain=1#L43

The IDs seem to be numbered by the code copy button feature. I have yet to figure out how the IDs are defined.

**Reply by SPyofgame** [reply]
If I am not wrong then there might be another bug that is related to this in that mkdocs.

By the way, can I ask that do Insiders have a discord or any communication site ?

**Reply by squidfunk** [reply]
Thanks for investigating – this is likely an inconsistency between the community version and Insiders due to the latest addition of the code line highlight feature. Could you please [create a documentation issue](https://github.com/squidfunk/mkdocs-material/issues/new?assignees=&labels=documentation&template=02-report-a-docs-issue.yml)? I'll look into it then.

**Reply by squidfunk** [reply]
> By the way, can I ask that do Insiders have a discord or any communication site ?

All communication should take place here on this repository and, for quick questions, on our [Gitter channel](https://app.gitter.im/#/room/#squidfunk_mkdocs-material:gitter.im).

**Reply by ile-24556** [reply]
Added the issue #5213.

**Reply by SPyofgame** [reply]
Thanks

**Comment by squidfunk** [comment, marked as answer]
This problem was related to an inconsistency of the community edition and Insiders. Reported in #5213 and fixed in https://github.com/squidfunk/mkdocs-material/commit/5f02d02dc44027212365040dbcd61ff833ecd8ae.
